internal/metrics/recorder.go

```diff
@@ -83,7 +95,56 @@ func NewRecorder(register bool) *Recorder {
 		addonsCount:           addonsCount,
 		addonOperatorPaused:   addonOperatorPaused,
 		ocmAPIRequestDuration: ocmAPIReqDuration,
+		addonHealthInfo:       addonHealthInfo,
+	}
+}
+
+func (r *Recorder) RecordAddonHealthInfo(addonStatus addonsv1alpha1.AddonStatus,
+	addonName, csvName, clusterID string) {
```

Comment:
Any reason for not handing in the Addon instance completely?
e.g. `addon *addonsv1alpha1.Addon`

Reply:
No specific reason :) Just that it would have to copy the entire status object into the scope of the function, rather than just the 4 fields it required. Doesn't make much noticeable difference in terms of performance though :D